packages/battery_plus/lib/battery_plus.dart

```diff
@@ -4,12 +4,14 @@
 
 import 'dart:async';
 
-import 'package:battery_plus_platform_interface/battery_plus_platform_interface.dart';
+import 'package:battery_plus/battery_plus_platform_interface.dart';
 
 // Export enums from the platform_interface so plugin users can use them directly.
-export 'package:battery_plus_platform_interface/battery_plus_platform_interface.dart'
+export 'package:battery_plus/battery_plus_platform_interface.dart'
     show BatteryState;
 
+export 'src/battery_plus_linux.dart';
```

Comment:
i wish we could do something like
```
linux:
  dartPluginClass: BatteryPlusLinux
  fileName: src/battery_plus_linux.dart
```
in `pubspec.yaml` to avoid this export but it would probably require changes in the flutter tool.

Reply:
I will check if that is possible

Reply:
I tried that, but I was faced with this error:

```
Launching lib/main.dart on Linux in debug mode...
ERROR: .dart_tool/flutter_build/dart_plugin_registrant.dart:20:9: Error: Undefined name 'BatteryPlusLinux'.
ERROR:         BatteryPlusLinux.registerWith();
ERROR:         ^^^^^^^^^^^^^^^^
Building Linux application...                                           
Exception: Build process failed
```

When I removed the `export src/battery_plus_linux.dart`

Reply:
Well, da*m. Thanks for trying anyway. :) Looks like there's some inconsistent `filename`/`fileName` handling in [flutter_tools/lib/src/platform_plugins.dart](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/platform_plugins.dart). Web uses `fileName` for Dart plugins, whereas Windows and Linux use `filename` but only for C++ plugins...